source/core/replica-set-high-availability.txt

```diff
@@ -15,29 +19,244 @@ Replica Set High Availability
    :depth: 1
    :class: singlecol
 
-:term:`Replica sets <replica set>` provide high availability using
-automatic :term:`failover`. Failover allows a :term:`secondary` member
-to become :term:`primary` if the current primary becomes unavailable.
+:term:`Replica sets <replica set>` use elections to select the 
+:term:`primary` for the cluster. Replica set elections support
+high availability by allowing clusters to automatically recover 
+normal operations after a primary failover event. Applications can
+continue to interact with the cluster during elections, mitigating
+the risk of downtime due to critical errors such as network 
+disruptions and hardware failure.
+
+Replica sets can trigger an election in response to a variety of events,
+such as:
+
+- Adding a new node to the replica set,
+- :method:`initiating a replica set <rs.initiate()>`,
+- performing replica set maintenance using methods such as :method:`rs.stepDown()` or :method:`rs.reconfig()`, and
+- the :term:`secondary` members losing connectivity to the primary for more than the configured :rsconf:`timeout <settings.electionTimeoutMillis>` (10 seconds by default).
+
+In the following diagram, the primary node unavailability
+exceeded the 
+:rsconf:`configured timeout <settings.electionTimeoutMillis>`.
+One of the remaining secondaries calls for an election in response to
+primary node availability. The cluster attempts to elects a new primary 
+and ensure continued write availability.
+
+.. include:: /images/replica-set-trigger-election.rst
+
+During an election the replica set can continue to serve read queries 
+if such queries are configured to 
+:ref:`run on secondaries <replica-set-read-preference>`. The
+replica set cannot process write operations until the election completes
+successfully. 
+
+.. include:: /includes/fact-retryable-writes-election.rst
+
+The median time in which a cluster may operate without a primary
+should not typically exceed 11.5 seconds, assuming default
+:rsconf:`replica configuration settings <settings>`. 
+This includes time required to mark the primary as
+:ref:`unavailable <replication-auto-failover>`.
+
+.. include:: /includes/fact-election-latency.rst
+
+If a majority of replica set members are inaccessible or unavailable,
+the remaining minority of replica set membrers cannot meet quorum for
+electing a new primary. If the primary node is among the remaining
+minority of replica set, it steps down and the cluster becomes 
+read-only. When a majority of replica set members are available, the 
+members can meet quurom for electing a new primary. Until that time, 
+the cluster can continue to serve read queries if such queries are 
+configured to run on secondaries. See :ref:`replica-set-read-preference`
+for complete documentation on secondary read operations.
```

Comment:
I think we can tighten this paragraph a bit, but may be taken care of if we tighten up the presentation of information a bit more on this page.

Also, isn't this kind of similar to the information regarding network partition?